runtime/compiler/codegen/J9CodeGenerator.cpp

```diff
@@ -780,23 +780,91 @@ J9::CodeGenerator::lowerTreeIfNeeded(
+         //
+         // Furthermore, for unresolved invokedynamic and invokehandle bytecodes, we create a call to linkToStatic.
+         // The appendix object in the invoke cache array entry could be NULL, which we cannot determine at compile
+         // time when the callSite/invokeCache table entries are unresolved. The VM would have to remove the appendix
+         // object, which would require knowing the number of args of the ROM method. To pass that information, we store to
+         // vmThread.floatTemp1 field. The name of the field is misleading, as it is an lconst/iconst being stored.
+         // If the linkToStatic call is not for an unresolved invokedynamic/invokehandle, then the JIT would not create a push
+         // of a null appendix object, so the VM would not need to do any stack adjustments. In those cases, -1 is stored.
+         // This is also done for linkToSpecial, as it shares the same handling mechanism in the VM.
          TR::Node * numArgsNode = NULL;
-         TR::Node * storeNode = NULL;
-         if (self()->comp()->target().is64Bit())
+         TR::Node * numArgSlotsNode = NULL;
+         TR::Node * tempSlotStoreNode = NULL;
+         TR::Node * floatTemp1StoreNode = NULL;
+         bool is64Bit = self()->comp()->target().is64Bit();
+         TR::ILOpCodes storeOpCode;
+         int32_t numParameterStackSlots = node->getSymbol()->castToResolvedMethodSymbol()->getNumParameterSlots();
+         if (is64Bit)
             {
-            numArgsNode = TR::Node::lconst(node, numParameterStackSlots);
-            storeNode = TR::Node::createStore(vmThreadTempSlotSymRef, numArgsNode, TR::lstore);
+            storeOpCode = TR::lstore;
+            numArgSlotsNode = TR::Node::lconst(node, numParameterStackSlots);
             }
          else
             {
-            numArgsNode = TR::Node::iconst(node, numParameterStackSlots);
-            storeNode = TR::Node::createStore(vmThreadTempSlotSymRef, numArgsNode, TR::istore);
+            storeOpCode = TR::istore;
+            numArgSlotsNode = TR::Node::iconst(node, numParameterStackSlots);
+            }
+         tempSlotStoreNode = TR::Node::createStore(self()->comp()->getSymRefTab()->findOrCreateVMThreadTempSlotFieldSymbolRef(),
+                                                   numArgSlotsNode,
+                                                   storeOpCode);
+         tempSlotStoreNode->setByteCodeIndex(node->getByteCodeIndex());
+         TR::TreeTop::create(self()->comp(), tt->getPrevTreeTop(), tempSlotStoreNode);
+
+         if (rm == TR::java_lang_invoke_MethodHandle_linkToStatic || rm ==  TR::java_lang_invoke_MethodHandle_linkToSpecial)
+            {
+            int32_t numArgs;
+            if (node->getSymbolReference()->getSymbol()->isDummyResolvedMethod())
+               {
+               // The last child of the call node is the iload for the memberName object, from which we can obtain the callSite index or cpIndex.
```

Comment:
You say `iload` but doesn't the code below assume an indirect load (shadow)?   The code below assumes the load has a first child, which implies an indirect load.

Can you make the comment clearer?